scripts/build_renewable_profiles.py

```diff
@@ -180,220 +179,158 @@
   reached.
 
 """
-import logging
-from _helpers import configure_logging
-
-import os
-import atlite
-
-import numpy as np
-import xarray as xr
-import pandas as pd
-import multiprocessing as mp
-import matplotlib.pyplot as plt
 import progressbar as pgb
-
-from scipy.sparse import csr_matrix, vstack
+import geopandas as gpd
+import xarray as xr
+import numpy as np
+import atlite
+import logging
 from pypsa.geo import haversine
-from vresutils import landuse as vlanduse
-from vresutils.array import spdiag
-
-logger = logging.getLogger(__name__)
-
-bounds = dx = dy = config = paths = gebco = clc = natura = None
+from shapely.geometry import LineString
+import time
 
+from _helpers import configure_logging
 
-def init_globals(bounds_xXyY, n_dx, n_dy, n_config, n_paths):
-    # Late import so that the GDAL Context is only created in the new processes
-    global gl, gk, gdal
-    import glaes as gl
-    import geokit as gk
-    from osgeo import gdal as gdal
+logger = logging.getLogger(__name__)
 
-    # global in each process of the multiprocessing.Pool
-    global bounds, dx, dy, config, paths, gebco, clc, natura
 
-    bounds = gk.Extent.from_xXyY(bounds_xXyY)
-    dx = n_dx
-    dy = n_dy
-    config = n_config
-    paths = n_paths
+if __name__ == '__main__':
+    if 'snakemake' not in globals():
+        from _helpers import mock_snakemake
+        snakemake = mock_snakemake('build_renewable_profiles', technology='solar')
+    configure_logging(snakemake)
+    pgb.streams.wrap_stderr()
+    paths = snakemake.input
+    nprocesses = snakemake.config['atlite'].get('nprocesses')
+    noprogress = not snakemake.config['atlite'].get('show_progress', True)
+    config = snakemake.config['renewable'][snakemake.wildcards.technology]
+    resource = config['resource'] # pv panel config / wind turbine config
+    correction_factor = config.get('correction_factor', 1.)
+    capacity_per_sqkm = config['capacity_per_sqkm']
+    p_nom_max_meth = config.get('potential', 'conservative')
 
-    if "max_depth" in config:
-        gebco = gk.raster.loadRaster(paths["gebco"])
-        gebco.SetProjection(gk.srs.loadSRS(4326).ExportToWkt())
+    if isinstance(config.get("corine", {}), list):
+        config['corine'] = {'grid_codes': config['corine']}
 
-    clc = gk.raster.loadRaster(paths["corine"])
-    clc.SetProjection(gk.srs.loadSRS(3035).ExportToWkt())
+    if correction_factor != 1.:
+        logger.info(f'correction_factor is set as {correction_factor}')
```

Comment:
Thank you! I remember this `correction_factor` bugging my when I started with PyPSA-EUR. :)